crypto/src/proofs/transparent.rs

```diff
@@ -56,63 +57,46 @@ impl SpendProof {
             return Ok(());
         }
 
-        // Note commitment integrity.
-        let s_component_transmission_key = self.note.transmission_key_s();
-        let note_commitment_test = note::commitment(
+        gadgets::note_commitment_integrity(
             self.note.note_blinding(),
             self.note.value(),
             self.note.diversified_generator(),
-            s_component_transmission_key,
-            self.note.clue_key(),
-        );
-
-        if self.note_commitment_proof.commitment() != note_commitment_test {
-            return Err(anyhow!("note commitment mismatch"));
-        }
+            self.note.transmission_key_s(),
+            self.note.clue_key().clone(),
+            self.note_commitment_proof.commitment(),
+        )?;
 
         // Merkle path integrity.
         self.note_commitment_proof
             .verify(anchor)
             .map_err(|_| anyhow!("merkle root mismatch"))?;
 
-        // Value commitment integrity.
-        if self.note.value().commit(self.v_blinding) != balance_commitment {
-            return Err(anyhow!("value commitment mismatch"));
-        }
+        gadgets::value_commitment_integrity(
```

Comment:
Now that we've gone from "value commitment" to "balance commitment", should this gadget be `balance_commitment_integrity`?